common/src/db/query.rs

```diff
@@ -422,35 +531,53 @@ impl From<Filter> for ConditionExpression {
 }
 
 /////////////////////////////////////////////////////////////////////////
-// FromStr impls
+// Sort
 /////////////////////////////////////////////////////////////////////////
 
-impl FromStr for Operator {
-    type Err = Error;
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
-        match s {
-            "=" => Ok(Operator::Equal),
-            "!=" => Ok(Operator::NotEqual),
-            "~" => Ok(Operator::Like),
-            "!~" => Ok(Operator::NotLike),
-            ">" => Ok(Operator::GreaterThan),
-            ">=" => Ok(Operator::GreaterThanOrEqual),
-            "<" => Ok(Operator::LessThan),
-            "<=" => Ok(Operator::LessThanOrEqual),
-            "|" => Ok(Operator::Or),
-            "&" => Ok(Operator::And),
-            _ => Err(Error::SearchSyntax(format!("Invalid operator: '{s}'"))),
+struct Sort {
+    field: ColumnRef,
+    order: Order,
+}
+
+impl Sort {
+    fn parse(s: &str, columns: &Columns) -> Result<Self, Error> {
+        let (field, order) = match s.split(':').collect::<Vec<_>>()[..] {
+            [f] => (f, String::from("asc")),
+            [f, dir] => (f, dir.to_lowercase()),
```

Comment:
to address my own ignorance - what is **dir** in this context ?

Reply:
direction. sort fields can have an optional suffix of either `:asc` or `:desc`. default is `asc` if omitted.

Reply:
ug, yes of course ...